thirdparty/protobuf/CMakeLists.txt

```diff
@@ -0,0 +1,125 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+#------------------------------------------------------------------------------
+# Configure and install Google Protobuf ...
+#------------------------------------------------------------------------------
+
+set(PUSH_CMAKE_INTERPROCEDURAL_OPTIMIZATION_RELEASE "${CMAKE_INTERPROCEDURAL_OPTIMIZATION_RELEASE}")
+set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_RELEASE OFF)
+
+if (MSVC)
+    set(protobuf_MSVC_STATIC_RUNTIME OFF CACHE BOOL "")
+endif()
+
+if(CMAKE_CROSSCOMPILING)
+    find_program(SYSTEM_PROTOC protoc PATHS ENV PATH)
+
+    if(SYSTEM_PROTOC)
+        execute_process(
+            COMMAND ${SYSTEM_PROTOC} --version
+            OUTPUT_VARIABLE PROTOC_VERSION
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+
+        string(REPLACE " " ";" PROTOC_VERSION ${PROTOC_VERSION})
+        list(GET PROTOC_VERSION -1 PROTOC_VERSION)
+
+        message("Detected system protoc version: ${PROTOC_VERSION}")
+    else()
+        message(FATAL_ERROR "System Protobuf is needed while cross-compiling")
+    endif()
+
+    set(protobuf_BUILD_PROTOC_BINARIES OFF CACHE BOOL "Build libprotoc and protoc compiler" FORCE)
+endif()
+
+if (CMAKE_GENERATOR STREQUAL "Ninja")
+    set(MAKE_UTIL make)
+else()
+    set(MAKE_UTIL $(MAKE))
+endif()
+
+set(protobuf_BUILD_TESTS OFF CACHE BOOL "Build tests")
+set(protobuf_WITH_ZLIB OFF CACHE BOOL "Build with zlib support")
+
+if (NOT BUILD_STANDALONE_STATIC)
+    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/protobuf/cmake EXCLUDE_FROM_ALL)
+    get_directory_property(protobuf_VERSION DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/protobuf/cmake DEFINITION protobuf_VERSION)
+endif()
+if (USE_STATIC_PROTOBUF)
+    include(FetchContent)
+    FetchContent_Declare(
+            ext_protobuf_static
+            GIT_REPOSITORY ${CMAKE_CURRENT_SOURCE_DIR}/protobuf
+    )
```

Comment:
I don't understand what happens here:
1. We have add_subdirectory
2. We have FetchContent_Declare
3. We have custom command

Why do we need 3 of them? The original idea was add_subdirectory creates protobuf for ONNX importer, FetchContent_Declare for PDPD

Reply:
Now, the FetchContent for PDPD uses the local protobuf submodule as a local repository to avoid download.
It will create a protobuf copy for PDPD.